src/rust/zh.manhuagui/src/parser.rs

```diff
@@ -0,0 +1,351 @@
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: format!("{}/comic/{}", BASE_URL, manga_id), //`${this.baseUrl}/comic/${mangaId}`;,
+			categories: vec![],
+			status: MangaStatus::Completed,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Scroll,
+		};
+		mangas.push(manga);
+	}
+
+	let mut has_next: bool = false;
+	for page in html.select("#AspNetPagerResult > a").array() {
+		let page_node = match page.as_node() {
+			Ok(node) => node,
+			Err(_) => continue,
+		};
+		if page_node.text().read() == "尾页" {
+			has_next = true;
+			break;
+		}
+	}
+
+
+	Ok(MangaPageResult {
+		manga: mangas,
+		has_more: has_next,
+	})
+}
+
+pub fn parse_manga_details(html: Node, manga_id: String) -> Result<Manga> {
+	let title = html.select(".book-title > h1").text().read();
+	let author = html
+      .select(
+        "body > div.w998.bc.cf > div.fl.w728 > div.book-cont.cf > div.book-detail.pr.fr > ul > li:nth-child(2) > span:nth-child(2) > a:nth-child(2)"
+      )
+      .text()
+	  .read();
+	let desc = html.select("#intro-cut").text().read();
+	let image = format!("https://cf.hamreus.com/cpic/b/{}.jpg", manga_id);
+	let url = format!("https://www.manhuagui.com/comic/{}/", manga_id);
+
+	let manga = Manga {
+		id: manga_id,
+		cover: image,
+		title,
+		author: author.clone(),
+		artist: author,
+		description: desc,
+		url,
+		categories: vec![],
+		status: MangaStatus::Ongoing,
+		nsfw: MangaContentRating::Safe,
+		viewer: MangaViewer::Scroll,
+	};
+
+	Ok(manga)
+}
+
+pub fn get_chapter_list(html: Node) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	let mut index = 1.0;
+
+	let mut div = html.clone();
+	let hidden = html.html().read().contains("__VIEWSTATE");
+	if hidden {
+		let compressed = html.select("#__VIEWSTATE").attr("value").read();
+		let decompressed =
+			String::from_utf16(&decompress_from_base64(compressed.as_str()).unwrap_or_default()).unwrap_or_default();
+		div = Node::new_fragment(decompressed.as_bytes()).unwrap_or(div);
+	}
+
+	for element in div.select(".chapter-list").array() {
+		let chapt_list_div = match element.as_node() {
+			Ok(node) => node,
+			Err(_) => continue,
+		};
+
+		for ul_ref in chapt_list_div.select("ul").array() {
+			let ul = match ul_ref.as_node() {
+				Ok(node) => node,
+				Err(_) => continue,
+			};
+
+			for li_ref in ul.select("li").array() {
+				let elem = match li_ref.as_node() {
+					Ok(node) => node,
+					Err(_) => continue,
+				};
+
+				let url = elem.select("a").attr("href").read();
+				let id = url.clone().replace("/comic/", "").replace(".html", "");
+				let chapter_id = match id.split('/').last() {
+					Some(id) => String::from(id),
+					None => String::new(),
+				};
+				let title = elem.select("a").attr("title").read();
+				let mut ch = title
+					.clone()
+					.replace(['第', '话', '卷'], "")
+					.parse::<f32>()
+					.unwrap_or(index);
+				if title.contains('卷') {
+					ch += 99999.0;
```

Comment:
why do you do this?

Reply:
I think you should just leave the chapter number as is. I can't say I know what that means exactly, but the numbers shouldn't be large like that for whatever reason.

Reply:
I understand now, I can put -1.0 to chapter and put the real number to volume. thanks